stock_full_location_reservation/models/stock_move.py

```diff
@@ -13,6 +13,22 @@ class StockMove(models.Model):
     def _filter_full_location_reservation_moves(self):
         return self.filtered(lambda m: m.is_full_location_reservation)
 
+    def _action_cancel(self):
+        """
+        Override the method to do the unlink only as the super() do
+        the unreserve also.
+        """
+        # If we are already in a undo_full_location_reservation context,
+        # we return super() method.
+        if self.env.context.get("skip_undo_full_location_reservation"):
+            return super()._action_cancel()
+        new_self = self.with_context(skip_undo_full_location_reservation=True)
+        res = super(StockMove, new_self)._action_cancel()
+        full_location_moves = self._filter_full_location_reservation_moves()
+        if full_location_moves:
+            full_location_moves.unlink()
+        return res
```

Comment:
```suggestion
        # If we are already in a undo_full_location_reservation context,
        # we return super() method.
        skip_undo = self.env.context.get("skip_undo_full_location_reservation")
        self = self.with_context(skip_undo_full_location_reservation=True)
        res = super()._action_cancel()
        if not skip_undo:
            full_location_moves = self._filter_full_location_reservation_moves()
            if full_location_moves:
                full_location_moves.unlink()
        return res
```

Reply:
Why? It is working for me. 
But this is not a blocker. :-) I'm fine with `new_self`

Reply:
> Why? It is working for me.

Yes it works, but semantically it is not quite good. As self is not the same as the one at the function beginning. When reading code, you should be able to identify that 'self' is not what you expect to be.

Reply:
Ok yes makes sense. :-)